modin/engines/base/frame/data.py

```diff
@@ -854,20 +854,25 @@ def _fold(self, axis, func):
             self._column_widths,
         )
 
-    def filter_full_axis(self, axis, func):
+    def filter_full_axis(self, axis, func, invert_axis=None):
         """Filter data based on the function provided along an entire axis.
 
         Args:
             axis: The axis to filter over.
             func: The function to use for the filter. This function should filter the
                 data itself.
+            invert_axis: The flag to invert axis for computation of metadata.
```

Comment:
Can we just pass `axis=1` above instead of introducing a new flag in the internal API? Please let me know if there are any issues with this approach.

Reply:
We can't just pass `axis=1` for `unique` method because we need to build column partitions (it will be done with `axis=0`) in order to compute unique values correctly. Then, we need to recalculate new `index` and new `row_lengths` (it will be done with `axis=1`). So, if we pass just `axis=1`, row partitions will be built and unique values will be computed incorrectly.

Reply:
To avoid introducing a new flag for `filter_full_axis` we can add another method that will do half functionality of `filter_full_axis`. What do you think about it?

Reply:
I see the issue, you should be able to use `_apply_full_axis` instead. `filter_full_axis` is about filtering rows based on some state in the rows. Does that make sense?

Reply:
Oh, I forgot about `_apply_full_axis`. I replaced `filter_full_axis` call to `_apply_full_axis`. Thanks a lot!